### ISSUE TITLE
fix: SelectedOptionChanged called twice in FluentCombobox

### ISSUE THREAD

**Issue by nwoolls** [microsoft/fluentui-blazor#2077]
# 🐛 Bug Report

I am trying to handle scenarios where a user clears the value in a `FluentCombobox`, or types a value that is not in the list. I am using `SelectedOptionChanged` to do so. However, `SelectedOptionChanged` is called twice, once with a parameter of `null` and then again with the previously selected item, even though the rendered combo box does not have the previously selected item in it.

## 💻 Repro or Code Sample

```bash
dotnet new fluentblazor
```

```razor
@rendermode InteractiveServer

<FluentCombobox Items="_items"
                TOption="string"
                SelectedOptionChanged="x => HandleOriginProductChanged(x)"/>

@code {
    private List<string> _items = [ "Item 1", "Item 2", "Item 3" ];

    private void HandleOriginProductChanged(string? value)
    {
        Console.WriteLine($"Selected product: {value}");
    }
}
```

1. Run the above sample
2. Select "Item 1" from the combo
3. Select "Item 2" from the combo
4. Clear out the selected value (or type a value that does not exist in the list)

## 🤔 Expected Behavior

Console log:

```
Selected product: Item 1
Selected product: Item 2
Selected product: 
```

## 😯 Current Behavior

Console log:

```
Selected product: Item 1
Selected product: Item 2
Selected product: 
Selected product: Item 2
```

## 💁 Possible Solution

N/A - I may be willing to work on a fix time permitting.

## 🔦 Context

FWIW I saw a similar issue here that was fixed in 4.7.2, but I am still seeing the problem reported above:

* https://github.com/microsoft/fluentui-blazor/issues/1965

## 🌍 Your Environment

* OS & Device: MacOS
* Browser: Google Chrome, Apple Safari
* .NET and Fluent UI Blazor library Version: 8.0.5 and 4.7.2

**Comment by vnbaaij** [comment]
We are in the middle of changing the list components to inherit from `InputBase` (instead of `ComponentBase`). We will take this in as part of that work.

**Comment by vnbaaij** [comment]
Can confirm that this will be fixed with PR #2118
